packages/core/src/crawlers/crawler_commons.ts

```diff
@@ -303,6 +303,7 @@ export class RequestHandlerResult {
             setValue: async (key, value, options) => {
                 this.setKeyValueStoreChangedValue(idOrName, key, value, options);
             },
+            getPublicUrl: store.getPublicUrl,
```

Comment:
this needs to be bound i guess?
```suggestion
            getPublicUrl: store.getPublicUrl.bind(store),
```